modules/tensor_mechanics/doc/content/documentation/systems/UserObjects/GeneralizedPlaneStrainUserObject.md

```diff
@@ -1,19 +1,30 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
-
 # GeneralizedPlaneStrainUserObject
 
-!alert construction title=Undocumented Class
-The GeneralizedPlaneStrainUserObject has not been documented, if you would like to contribute to MOOSE by
-writing documentation, please see [/generate.md]. The content contained on this page explains
-the typical documentation associated with a MooseObject; however, what is contained is ultimately
-determined by what is necessary to make the documentation clear for users.
-
 !syntax description /UserObjects/GeneralizedPlaneStrainUserObject
 
+UserObject `GeneralizedPlaneStrainUserObject` calculates the values of residual and diagonal jacobian components for each given scalar out-of-plane strain variable.
+
+For a given scalar out-of-plane strain variable, the equilibrium condition condition in the out-of-plane direction (e.g. $z$-direction) is given as
+\begin{equation}
+	\int_{A}{\sigma_{zz}dA} = \bar{N}_{zz}
+\end{equation}
+
+Thus, the residual corresponding to the scalar out-of-plane strain variable is
+
+\begin{equation}
+	R = \int_{A}{\sigma_{zz}dA} - \bar{N}_{zz}
+\end{equation}
+
+and the corresponding diagonal jacobian is
+
+\begin{equation}
+	J = \frac{\partial R_{zz}}{\partial \epsilon_{zz}} = \int_{A}{\frac{\partial \sigma_{zz}}{\partial \epsilon_{zz}}dA} = \int_{A}{C_{2222}dA}
+\end{equation}
+
+A detailed description of generalized plane strain formulation can be found in [here](tensor_mechanics/generalized_plane_strain.md).
+
```

Comment:
Mention here that this is typically set up by the Action, and add a link to that documentation.